lib/database_plumber/leak_finder.rb

```diff
@@ -12,12 +12,13 @@ def self.inspect(options = {})
     def initialize(options)
       @ignored_models = (options[:ignored_models] || []) + IGNORED_AR_INTERNALS
       @ignored_adapters = options[:ignored_adapters] || []
+      @model_thresholds = options[:model_thresholds] || {}
     end
 
     def inspect
       filtered_models.each_with_object({}) do |model, results|
         records = count_for(model)
-        if records > 0
+        if records > threshold_for(model)
           results[model.to_s] = records
           mop_up(model)
```

Comment:
NAB: I'm a little iffy on the cleanup behaviour for a thresholded model, but I guess if a test run is seeded and we have polluted the seeds we should expect to have to re-apply them. There's no intelligent way to know which rows to delete unless we expand the gem to audit the DB content at the start of the run and that's problematic too.